crates/fiber/src/lib.rs

```diff
@@ -175,7 +197,12 @@ impl<'a, Resume, Yield, Return> Fiber<'a, Resume, Yield, Return> {
                 Err(y)
             }
             RunResult::Returned(r) => Ok(r),
-            RunResult::Panicked(payload) => std::panic::resume_unwind(payload),
+            RunResult::Panicked(_payload) => {
+                #[cfg(feature = "std")]
+                panic::resume_unwind(_payload);
+                #[cfg(not(feature = "std"))]
+                unreachable!()
+            }
```

Comment:
Could this be avoided by having a `#[cfg]` for std/unwind on the `Panicked` variant? That way we can statically provide that it's never constructed

Reply:
Done -- refactored a bit in `Suspend::execute()` to statically avoid constructing the panic case as well.